### PR TITLE
gnuplot: qt5 support

### PR DESCRIPTION
**Pull request by schoeps**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
@schoeps It's worth noting we're trying to transition all Qt-using formulae to using only Qt5: https://github.com/Homebrew/homebrew-core/issues/1705. This may make work both here and in https://github.com/Homebrew/homebrew-core/pull/3733 easier; you can remove the `with-qt` options and replace them with `with-qt5`.

**Comment by schoeps** [comment]
you recommend to mark `--with-qt` as deprecated and only support qt5?

**Comment by MikeMcQuaid** [comment]
Yep. You can just remove all `with-qt` options and references that aren't to Qt5 and perhaps do `deprecated_option "with-qt", "with-qt5"` which will migrate people over.

File: Formula/gnuplot.rb [hunk 1]
```diff
@@ -28,7 +28,8 @@ class Gnuplot < Formula
   deprecated_option "with-x" => "with-x11"
   deprecated_option "pdf" => "with-pdflib-lite"
   deprecated_option "wx" => "with-wxmac"
-  deprecated_option "qt" => "with-qt"
```
Comment (MikeMcQuaid) [inline Formula/gnuplot.rb]: Can change this to `deprecated_option "qt" => "with-qt5"`

File: Formula/gnuplot.rb [hunk 2]
```diff
@@ -47,23 +48,27 @@ class Gnuplot < Formula
...
   depends_on "pdflib-lite" => :optional
-  depends_on "qt" => :optional
+  depends_on "qt5" => :optional
   depends_on "wxmac" => :optional
   depends_on :tex => :optional
   depends_on :x11 => :optional
 
   def install
-    if build.with? "aquaterm"
+    if build.with?("aquaterm")
```
Comment (MikeMcQuaid) [inline Formula/gnuplot.rb]: Leave this as it was, thanks.

File: Formula/gnuplot.rb [hunk 3]
```diff
@@ -47,23 +48,27 @@ class Gnuplot < Formula
...
       # Add "/Library/Frameworks" to the default framework search path, so that an
       # installed AquaTerm framework can be found. Brew does not add this path
       # when building against an SDK (Nov 2013).
       ENV.prepend "CPPFLAGS", "-F/Library/Frameworks"
       ENV.prepend "LDFLAGS", "-F/Library/Frameworks"
     end
 
+    if build.with?("qt5")
+      # make sure qt5 is found by configure
+      ENV.prepend_path "PATH", Formula["qt5"].bin
```
Comment (MikeMcQuaid) [inline Formula/gnuplot.rb]: This is done automatically and shouldn't be necessary.

File: Formula/gnuplot.rb [hunk 4]
```diff
@@ -72,14 +77,14 @@ def install
       --with-readline=#{Formula["readline"].opt_prefix}
     ]
 
-    args << "--with-pdf=#{pdflib}" if build.with? "pdflib-lite"
+    args << "--with-pdf=#{pdflib}" if build.with?("pdflib-lite")
```
Comment (MikeMcQuaid) [inline Formula/gnuplot.rb]: Leave this as it was, thanks.

File: Formula/gnuplot.rb [hunk 5]
```diff
@@ -72,14 +77,14 @@ def install
...
 
-    if build.without? "wxmac"
+    if build.without?("wxmac")
```
Comment (MikeMcQuaid) [inline Formula/gnuplot.rb]: Leave this as it was, thanks.

File: Formula/gnuplot.rb [hunk 6]
```diff
@@ -72,14 +77,14 @@ def install
...
       args << "--disable-wxwidgets"
-      args << "--without-cairo" if build.without? "cairo"
+      args << "--without-cairo" if build.without?("cairo")
```
Comment (MikeMcQuaid) [inline Formula/gnuplot.rb]: Leave this as it was, thanks.

File: Formula/gnuplot.rb [hunk 7]
```diff
@@ -72,14 +77,14 @@ def install
...
     end
 
-    if build.with? "qt"
+    if build.with?("qt") || build.with?("qt5")
```
Comment (MikeMcQuaid) [inline Formula/gnuplot.rb]: Can remove `build.with?("qt")` and make this become `if build.with? "qt5"`

File: Formula/gnuplot.rb [hunk 8]
```diff
@@ -89,11 +94,11 @@ def install
     # or it halts in the middle of the build for user-interactive resolution.
     # Per upstream: "--with-tutorial is horribly out of date."
     args << "--without-tutorial"
-    args << "--without-lua" if build.without? "lua"
+    args << "--without-lua" if build.without?("lua")
```
Comment (MikeMcQuaid) [inline Formula/gnuplot.rb]: Leave this as it was, thanks.

File: Formula/gnuplot.rb [hunk 9]
```diff
@@ -89,11 +94,11 @@ def install
...
     args << ((build.with? "aquaterm") ? "--with-aquaterm" : "--without-aquaterm")
     args << ((build.with? "x11") ? "--with-x" : "--without-x")
 
-    if build.with? "tex"
+    if build.with?("tex")
```
Comment (MikeMcQuaid) [inline Formula/gnuplot.rb]: Leave this as it was, thanks.

File: Formula/gnuplot.rb [hunk 10]
```diff
@@ -108,7 +113,7 @@ def install
   end
 
   def caveats
-    if build.with? "aquaterm"
+    if build.with?("aquaterm")
```
Comment (MikeMcQuaid) [inline Formula/gnuplot.rb]: Leave this as it was, thanks.

File: Formula/gnuplot.rb (hunk 3 above)
Comment (schoeps) [inline Formula/gnuplot.rb]: You are right. However, this is strange: I was playing with the formula for Octave 4.2 (which is the reason that I am interested in qt5) and it was always using the wrong moc if the path was not adapted.

Comment (MikeMcQuaid) [inline Formula/gnuplot.rb]: Yeh, Octave may have its own issues that need fixed 😀

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request merged**